File: schemas/notifications-system/drizzle/notification_feeds.ts
// notification_feeds: Named UI surfaces where notifications can appear (e.g., bell icon, activity tab, announcements banner).
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp } from "drizzle-orm/pg-core";

export const notificationFeeds = pgTable("notification_feeds", {
  id: uuid("id").primaryKey().defaultRandom(),
  name: text("name").notNull(), // Display name (e.g., "General", "Activity", "Announcements").
  slug: text("slug").unique().notNull(), // URL-safe identifier used in API calls: GET /feeds/general.
  description: text("description"), // Explain what this feed is for. Shown in admin UI.
  createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  updatedAt: timestamp("updated_at", { withTimezone: true })
    .notNull()
    .defaultNow()
    .$onUpdateFn(() => new Date()),
});

File: schemas/notifications-system/spacetimedb/notification_feeds.rs
// notification_feeds: Named UI surfaces where notifications can appear (e.g., bell icon, activity tab, announcements banner).
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = notification_feeds, public)]
pub struct NotificationFeed {
    #[primary_key]
    pub id: String, // UUID

    pub name: String, // Display name (e.g., "General", "Activity", "Announcements").

    #[unique]
    pub slug: String, // URL-safe identifier (e.g., "general", "activity"). Used in API calls: GET /feeds/general.

    pub description: Option<String>, // Explain what this feed is for. Shown in admin UI.

    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
